src/tokens/saga.ts

```diff
@@ -309,6 +329,54 @@ export function* watchAccountFundedOrLiquidated() {
   }
 }
 
+export async function fetchImportedTokenBalances(
+  address: Address,
+  importedTokens: StoredTokenBalances,
+  knownTokenBalances: Record<string, FetchedTokenBalance>
+) {
+  try {
+    const importedTokensList = Object.values(importedTokens)
+    const importedTokensWithBalance: StoredTokenBalances = {}
+
+    const balanceRequests = importedTokensList.map(async (importedToken) => {
+      try {
+        if (!importedToken) {
+          return
+        }
+
+        let fetchedBalance
+        if (knownTokenBalances[importedToken.tokenId]) {
+          fetchedBalance = knownTokenBalances[importedToken.tokenId].balance
+        } else {
+          const contract = getContract({
+            abi: erc20.abi,
+            address: importedToken!.address as Address,
+            client: {
+              public: publicClient[networkIdToNetwork[importedToken.networkId]],
+            },
+          })
+          fetchedBalance = (await contract.read.balanceOf([address])).toString()
+        }
+
+        const balance = new BigNumber(fetchedBalance).shiftedBy(-importedToken.decimals).toFixed()
+
+        importedTokensWithBalance[importedToken.tokenId] = {
+          ...importedToken,
+          balance,
+        }
+      } catch (error) {
+        Logger.error(TAG, 'Error fetching imported token balance', error)
+      }
+    })
+
+    await Promise.all(balanceRequests)
+    return importedTokensWithBalance
+  } catch (error) {
+    Logger.error(TAG, 'Error fetching imported tokens balances', error)
+    return {}
```

Comment:
now that you've added a try/catch inside each balance promise, does this overall try/catch still need to be here? i don't see anything else that can throw?

Reply:
Oops, sorry, yes I can remove this try/catch